### ISSUE TITLE
repack helper script does not work with signed executables on macOS

### ISSUE THREAD

**Issue by ntamas** [dashingsoft/pyarmor#547]
**Describe the bug**

The `repack.py` helper script does not work with recent versions of PyInstaller (I tried PyInstaller 4.4) on macOS. This is because PyInstaller started producing executables signed with a dummy code signature. The dummy code signature is required for Apple Silicon chips where each executable must be signed, otherwise the system won't execute it.

The signature breaks `repack.py` in two ways:

1. When `repack.py` scans the input file for the PyInstaller CArchive, it looks for the PyInstaller magic bytes at the end of the executable only. This will not work because the end of the executable now contains the code signature.
2. Even if `repack.py` is patched to scan larger and larger chunks of the input file from the end when looking for the CArchive (as PyInstaller does it now in its own code), the code signature must be stripped from the executable because modifying the CArchive would break the code signature anyway.

A workaround for the time being is to call the `codesign` CLI tool with the `--remove-signature` option before calling `repack.py`, but it would be great if `repack.py` could strip the signature on its own and print a warning to the user that the repacked executable is unsigned even if the original executable was signed. (Even better if it could re-sign the executable with the same identity somehow).

**To Reproduce**

PyArmor 6.7.3, purchased.
Python 3.9, macOS.

Call `python repack.py -p some_folder/obf some_pyinstaller_bundle` with a PyInstaller bundle that has a code signature at the end. You will see something like this:

```
...
1826 INFO: Patch PYZ done
1852 INFO: Appending runtime files to archive
1852 INFO: Add "build/pyarmor/obf/pytransform.so"
1852 INFO: Append runtime files OK
1879 INFO: Repacking EXE "some_pyinstaller_bundle"
1883 ERROR: some_pyinstaller_bundle is not a valid CArchive
```

**Comment by ntamas** [comment]
FWIW, it looks like PyInstaller includes a `remove_signature_from_binary()` function so you can try using that instead of calling `codesign` directly.

**Comment by jondy** [comment]
I'll check it

**Comment by jondy** [comment]
Fixed it by this commit b598a5e5cfadcee70840cb1baa06440a72d6809f

https://pyarmor.readthedocs.io/en/latest/change-logs.html#dev

Your issue report is excellent, I love it.

**Comment by ntamas** [comment]
Awesome, thanks for the quick response! I'll try it soon and get back to you if anything is amiss.
